### PR TITLE
Fix metric instrument name validation

### PR DESCRIPTION
**Pull request by lzchen**
Dash should be an underscore.

### CODE REVIEW

File: opentelemetry-sdk/src/metrics/meter.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ use crate::metrics::{
 const INSTRUMENT_NAME_MAX_LENGTH: usize = 255;
 // maximum length of instrument unit name
 const INSTRUMENT_UNIT_NAME_MAX_LENGTH: usize = 63;
-const INSTRUMENT_NAME_ALLOWED_NON_ALPHANUMERIC_CHARS: [char; 4] = ['-', '.', '-', '/'];
+const INSTRUMENT_NAME_ALLOWED_NON_ALPHANUMERIC_CHARS: [char; 4] = ['_', '.', '-', '/'];
```
Comment (cijothomas) [inline opentelemetry-sdk/src/metrics/meter.rs:28]: https://github.com/open-telemetry/opentelemetry-rust/issues/1265 This has some unit-tests added, wondering why the unittests did not caught this?

**Review by lalitb — APPROVED** [review]

File: opentelemetry-sdk/src/metrics/meter.rs (hunk 1 above)
Comment (lalitb) [reply opentelemetry-sdk/src/metrics/meter.rs:28]: I think the existing unit-test doesn't catch this. Good to add one.

Comment (TommyCpp) [reply opentelemetry-sdk/src/metrics/meter.rs:28]: +1 for unit tests

**Review by cijothomas — APPROVED** [review]

**Comment by ymgyt** [comment]
sorry about the misss in the move validation, and thanks for fix

File: opentelemetry-sdk/src/metrics/meter.rs (hunk 1 above)
Comment (cijothomas) [reply opentelemetry-sdk/src/metrics/meter.rs:28]: https://github.com/open-telemetry/opentelemetry-rust/issues/1219 is opened to track adding tests. Resolving this conversation and would suggest to merge this PR with the fix.

**Pull request merged**